File: src/main/java/com/github/simkuenzi/cronext/DayOfWeekField.java
package com.github.simkuenzi.cronext;

import java.time.LocalDateTime;

class DayOfWeekField implements CronField {
    private CronValue value;

    DayOfWeekField(CronValue value) {
        this.value = value;
    }

    @Override
    public boolean matches(LocalDateTime dateTime) {
        return value.matches(dateTime.getDayOfWeek().getValue() - 1, new Range(0, 6));
    }
}

File: src/main/kotlin/com/github/simkuenzi/cronext/DayOfWeekField.kt
package com.github.simkuenzi.cronext

import java.time.LocalDateTime

class DayOfWeekField(private val value: CronValue) : CronField {
    override fun matches(dateTime: LocalDateTime) = value.matches(dateTime.dayOfWeek.value - 1, 0..6)
}